app/src/org/commcare/utils/FormUploadUtil.java

```diff
@@ -204,17 +206,35 @@ private static FormUploadResult submitEntity(List<MultipartBody.Part> parts, Str
             return FormUploadResult.FULL_SUCCESS;
         } else if (responseCode == 401) {
             return FormUploadResult.AUTH_FAILURE;
+        } else if (responseCode == 406) {
+            return processActionableFaiure(response);
         } else if (responseCode == 422) {
             return handleProcessingFailure(response.errorBody().byteStream());
         } else {
             return FormUploadResult.FAILURE;
         }
     }
 
+    private static FormUploadResult processActionableFaiure(Response<ResponseBody> response) {
+        String message;
+        try {
+            JSONObject errorKeyAndDefault = new JSONObject(response.errorBody().string());
+            message = Localization.getWithDefault(
```

Comment:
I'm a bit confused by this line. Does HQ send down a _locale key_ with the error? That seems... really weird to me as a mechanism.

Reply:
@ctsims Yeah thats whats happening currently. HQ sends a locale key along with a default message.